pkg/resource/service.go

```diff
@@ -793,7 +793,8 @@ func (r *ServiceResource) Update(ctx context.Context, req resource.UpdateRequest
 	}
 	if plan.NumReplicas != state.NumReplicas {
 		scalingChange = true
-		if !plan.NumReplicas.IsNull() {
+		// Only update `num_replicas` on production tier.
+		if !plan.NumReplicas.IsNull() && plan.Tier.ValueString() == api.TierProduction {
```

Comment:
One day we might have a third tier or we might allow changing replicas for the dev tier.
I would prefer to keep num replicas editable in principle (a.k.a. not tied to a specific tier).

Why don't we simply extend the check as

```
if !plan.NumReplicas.IsNull() && !plan.NumReplicas.IsUnknown() {
```

this way we basically delegate num_replicas to the server side and we control which tiers can set the num_replicas by using validation like we do on line 362